src/ripple/app/main/DBInit.h

```diff
@@ -114,6 +114,54 @@ inline constexpr std::array<char const*, 8> TxDBInit{
 
 ////////////////////////////////////////////////////////////////////////////////
 
+// The Ledger Meta database maps ledger hashes to shard indexes
+inline constexpr auto LgrMetaDBName{"ledger_meta.db"};
+
+inline constexpr std::array LgrMetaDBPragma
+{
+    "PRAGMA page_size=4096;", "PRAGMA journal_size_limit=1582080;",
+        "PRAGMA max_page_count=2147483646;",
+#if (ULONG_MAX > UINT_MAX) && !defined(NO_SQLITE_MMAP)
+        "PRAGMA mmap_size=17179869184;"
+#endif
+};
+
+inline constexpr std::array<char const*, 3> LgrMetaDBInit{
+    {"BEGIN TRANSACTION;",
+
+     "CREATE TABLE IF NOT EXISTS LedgerMeta (        \
+        LedgerHash  CHARACTER(64) PRIMARY KEY,       \
+        ShardIndex  INTEGER                          \
```

Comment:
This is just a key-value table. Why not use NuDB instead of SQLite?

Reply:
Seemed like the natural choice since it's being used in a SQLite interface class. Are there significant benefits to using NuDB instead?

Reply:
In general, we are trying to minimize the SQLite dependency in rippled. So I feel like if we can use the same database that the shards themselves are being stored in, that would be ideal.